src/store.rs

```diff
@@ -96,80 +80,51 @@ use crate::utils::JsonBuilder;
 /// the store.
 #[derive(Clone, Debug)]
 pub struct Store {
-    /// The database.
-    db: sled::Db,
+    /// The base path for the store.
+    path: PathBuf,
 }
 
 impl Store {
-    /// Creates a new store based on configuration information.
-    pub fn new(db: sled::Db) -> Self {
-        Store { db }
-    }
-
-    /// Start a validation run with the store.
-    pub fn start(&self) -> Run {
-        Run::new(self)
+    /// Returns the base path for the given config.
+    fn create_base_dir(config: &Config) -> Result<PathBuf, Failed> {
+        // We are using "stored" since store was foolishly used in 0.9.0 for
+        // the database.
+        let path = config.cache_dir.join("stored");
+        if let Err(err) = fs::create_dir_all(&path) {
+            error!(
+                "Failed to create store directory {}: {}",
+                path.display(), err
+            );
+            return Err(Failed)
+        }
+        Ok(path)
     }
 
-    /// Cleans up the store.
+    /// Initializes the store without creating a value.
     ///
-    /// All publication points that have an expired manifest will be removed.
-    /// RRDP repositories that have no more publication points are removed,
-    /// too.
+    /// Ensures that the base directory exists and creates it of necessary.
     ///
-    /// The method also triggers a cleanup of the collector via the provided
-    /// collector cleanup object. All RRDP repositories and rsync modules that
-    /// have still non-expired publication points will be registered to be
-    /// retained with the collector cleanup and then a cleaning run is
-    /// started.
-    pub fn cleanup(
-        &self,
-        mut collector: Option<collector::Cleanup>,
-    ) -> Result<(), Failed> {
-        // Cleanup RRDP repositories
-        for name in self.db.tree_names() {
-            if let Some(names) = TreeNames::from_manifest_tree_name(&name) {
-                if names.repository_name == b"rsync" {
-                    continue
-                }
-
-                let uri = match uri::Https::from_slice(
-                    names.repository_name
-                ) {
-                    Ok(uri) => uri,
-                    Err(_) => {
-                        names.drop_trees(&self.db)?;
-                        continue
-                    }
-                };
-
-                if Repository::new(self, &names, true)?.cleanup_rrdp()? {
-                    if let Some(collector) = collector.as_mut() {
-                        collector.retain_rrdp_repository(&uri);
-                    }
-                }
-                else {
-                    names.drop_trees(&self.db)?;
-                }
-            }
-        }
-
-        // Cleanup rsync modules
-        Repository::new(
-            self, &TreeNames::rsync(), false,
-        )?.cleanup_rsync(&mut collector)?;
+    /// The function is called implicitly by [`new`][Self::new].
```

Comment:
Technically, yes, but all that `init` does is call that function. There is a separate `create_base_dir` because `new` needs the path whereas `init` doesn’t care, so we can’t really call `init` from `new` directly. Smoke and mirrors …

Reply:
It added a comment-comment justifying the comment.